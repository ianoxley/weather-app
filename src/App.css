* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html, body {
  height: 100%;
}

.app {
  display: grid;
  grid-template-rows: 150px 150px 1fr;
  min-height: 100%;
}

header {
  background-color: #222;
  color: white;
  grid-row: 1;
  height: 150px;
  padding: 20px;
}

.app-title {
  font-size: 1.5em;
  text-align: center;
}

ol {
  list-style: none;
}

#tabs {
  display: grid;
  grid-template-columns: 20% 20% 20% 20% 20%;
  grid-auto-flow: row;
  grid-row: 2;
}

#tabs > li:first-child {
  grid-column: 1;
}

#tabs > li:last-child {
  grid-column: 5;
}

.tab {
  align-self: center;
  justify-self: center;
}

#tab-panels {
  align-self: center;
  grid-row: 3;
  justify-self: center;
}

#tab-panels > li {
  display: none;
}

#tab-panels > li:target {
  display: block;
}

