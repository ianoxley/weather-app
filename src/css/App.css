* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html, body {
  height: 100%;
}

.app {
  display: grid;
  grid-template-rows: 150px auto 1fr;
  min-height: 100%;
}

ol {
  list-style: none;
}

