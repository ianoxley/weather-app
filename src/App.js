import React, { Component } from 'react';
import Header from './header';
import './css/App.css';

const apiKey = process.env.REACT_APP_API_KEY;
const location = 'Newcastle,gb';
const units = 'metric';

const url = `https://api.openweathermap.org/data/2.5/forecast?q=${location}&appid=${apiKey}&units=${units}`;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      data: null
    };
  }

  componentDidMount() {
    fetch(url)
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            data: result
          });
          if (window.history && window.history.replaceState) {
            window.history.replaceState(undefined, 'Today', '#day-1');
          }
        },
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      );
  }

  render() {
    const { isLoaded, error, data } = this.state;

    if (error) {
      return (
        <div>Error: { error.message }</div>
      );
    } else if (!isLoaded) {
      return (
        <div>Loading&hellip;</div>
      );
    } else {
      const { city, list } = data;
      return (
        <main className="app">
          <Header
            location={ city.name }
            currentTemperature={ list[0].main.temp }
            description={ list[0].weather[0].main }
            icon={ list[0].weather[0].icon }
          />
          <ol id="tabs">
            <li className="tab">
              <a href="#day-1">Day 1</a>
            </li>
            <li className="tab">
              <a href="#day-2">Day 2</a>
            </li>
            <li className="tab">
              <a href="#day-3">Day 3</a>
            </li>
            <li className="tab">
              <a href="#day-4">Day 4</a>
            </li>
            <li className="tab">
              <a href="#day-5">Day 5</a>
            </li>
          </ol>
          <ol id="tab-panels">
            <li id="day-1" className="tab-panel">
              <div>Content Day 1</div>
            </li>
            <li id="day-2" className="tab-panel">
              <div>Content Day 2</div>
            </li>
            <li id="day-3" className="tab-panel">
              <div>Content Day 3</div>
            </li>
            <li id="day-4" className="tab-panel">
              <div>Content Day 4</div>
            </li>
            <li id="day-5" className="tab-panel">
              <div>Content Day 5</div>
            </li>
          </ol>
        </main>
      );
    }
  }
}

export default App;
