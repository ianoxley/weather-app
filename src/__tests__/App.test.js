import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from '../App';

describe('App', () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() => {
      return new Promise((resolve, reject) => {
        resolve({
          ok: true,
          json() {
            return {
              city: {
                name: 'Newcastle'
              },
              list: [
                {
                  main: {
                    temp: 17
                  },
                  weather: [
                    {
                      main: 'Clear',
                      icon: '01n'
                    }
                  ]
                }
              ]
            }
          }
        });
      });
    });
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('constructor', () => {
    it('sets up the default state', () => {
      const wrapper = shallow(<App />);
      const state = wrapper.state();
      expect(state.isLoaded).toBe(false);
      expect(state.error).toBeNull();
      expect(state.data).toBeNull();
    });
  });

  describe('componentDidMount', () => {
    let spy;

    afterEach(() => {
      spy.mockClear();
    });

    // TODO: fix this test
    xit('triggers #day-1 on first load', () => {
      spy = jest.spyOn(window.history, 'replaceState');
      const wrapper = mount(<App />);
      expect(spy).toHaveBeenCalledWith(undefined, 'Today', '#day-1');
    });
  });

  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App />);
    });

    describe('when error is not null', () => {
      it('renders the error message', () => {
        wrapper.setState({
          error: {
            message: 'An error occurred'
          }
        });
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('when isLoaded is false', () => {
      it('renders the loading message', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('when isLoaded is true', () => {
      it('renders the data', () => {
        wrapper.setState({
          isLoaded: true,
          data: {
            city: {
              name: 'Newcastle'
            },
            list: [
              {
                main: {
                  temp: 17
                },
                weather: [
                  {
                    main: 'Clear',
                    icon: '01n'
                  }
                ]
              }
            ]
          }
        });

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
