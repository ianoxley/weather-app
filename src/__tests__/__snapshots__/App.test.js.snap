// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render when error is not null renders the error message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Error: ",
        "An error occurred",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Error: ",
      "An error occurred",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Error: ",
          "An error occurred",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Error: ",
        "An error occurred",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App render when isLoaded is false renders the loading message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading…",
    },
    "ref": null,
    "rendered": "Loading…",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading…",
      },
      "ref": null,
      "rendered": "Loading…",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App render when isLoaded is true renders the data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          currentTemperature={17}
          description="Clear"
          icon="01n"
          location="Newcastle"
/>,
        <ol
          id="tabs"
>
          <li
                    className="tab"
          >
                    <a
                              href="#day-1"
                    >
                              Day 1
                    </a>
          </li>
          <li
                    className="tab"
          >
                    <a
                              href="#day-2"
                    >
                              Day 2
                    </a>
          </li>
          <li
                    className="tab"
          >
                    <a
                              href="#day-3"
                    >
                              Day 3
                    </a>
          </li>
          <li
                    className="tab"
          >
                    <a
                              href="#day-4"
                    >
                              Day 4
                    </a>
          </li>
          <li
                    className="tab"
          >
                    <a
                              href="#day-5"
                    >
                              Day 5
                    </a>
          </li>
</ol>,
        <ol
          id="tab-panels"
>
          <li
                    className="tab-panel"
                    id="day-1"
          >
                    <div>
                              Content Day 1
                    </div>
          </li>
          <li
                    className="tab-panel"
                    id="day-2"
          >
                    <div>
                              Content Day 2
                    </div>
          </li>
          <li
                    className="tab-panel"
                    id="day-3"
          >
                    <div>
                              Content Day 3
                    </div>
          </li>
          <li
                    className="tab-panel"
                    id="day-4"
          >
                    <div>
                              Content Day 4
                    </div>
          </li>
          <li
                    className="tab-panel"
                    id="day-5"
          >
                    <div>
                              Content Day 5
                    </div>
          </li>
</ol>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentTemperature": 17,
          "description": "Clear",
          "icon": "01n",
          "location": "Newcastle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="tab"
>
              <a
                            href="#day-1"
              >
                            Day 1
              </a>
</li>,
            <li
              className="tab"
>
              <a
                            href="#day-2"
              >
                            Day 2
              </a>
</li>,
            <li
              className="tab"
>
              <a
                            href="#day-3"
              >
                            Day 3
              </a>
</li>,
            <li
              className="tab"
>
              <a
                            href="#day-4"
              >
                            Day 4
              </a>
</li>,
            <li
              className="tab"
>
              <a
                            href="#day-5"
              >
                            Day 5
              </a>
</li>,
          ],
          "id": "tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#day-1"
>
                Day 1
</a>,
              "className": "tab",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day 1",
                "href": "#day-1",
              },
              "ref": null,
              "rendered": "Day 1",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#day-2"
>
                Day 2
</a>,
              "className": "tab",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day 2",
                "href": "#day-2",
              },
              "ref": null,
              "rendered": "Day 2",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#day-3"
>
                Day 3
</a>,
              "className": "tab",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day 3",
                "href": "#day-3",
              },
              "ref": null,
              "rendered": "Day 3",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#day-4"
>
                Day 4
</a>,
              "className": "tab",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day 4",
                "href": "#day-4",
              },
              "ref": null,
              "rendered": "Day 4",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#day-5"
>
                Day 5
</a>,
              "className": "tab",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day 5",
                "href": "#day-5",
              },
              "ref": null,
              "rendered": "Day 5",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ol",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="tab-panel"
              id="day-1"
>
              <div>
                            Content Day 1
              </div>
</li>,
            <li
              className="tab-panel"
              id="day-2"
>
              <div>
                            Content Day 2
              </div>
</li>,
            <li
              className="tab-panel"
              id="day-3"
>
              <div>
                            Content Day 3
              </div>
</li>,
            <li
              className="tab-panel"
              id="day-4"
>
              <div>
                            Content Day 4
              </div>
</li>,
            <li
              className="tab-panel"
              id="day-5"
>
              <div>
                            Content Day 5
              </div>
</li>,
          ],
          "id": "tab-panels",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Content Day 1
</div>,
              "className": "tab-panel",
              "id": "day-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Content Day 1",
              },
              "ref": null,
              "rendered": "Content Day 1",
              "type": "div",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Content Day 2
</div>,
              "className": "tab-panel",
              "id": "day-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Content Day 2",
              },
              "ref": null,
              "rendered": "Content Day 2",
              "type": "div",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Content Day 3
</div>,
              "className": "tab-panel",
              "id": "day-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Content Day 3",
              },
              "ref": null,
              "rendered": "Content Day 3",
              "type": "div",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Content Day 4
</div>,
              "className": "tab-panel",
              "id": "day-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Content Day 4",
              },
              "ref": null,
              "rendered": "Content Day 4",
              "type": "div",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Content Day 5
</div>,
              "className": "tab-panel",
              "id": "day-5",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Content Day 5",
              },
              "ref": null,
              "rendered": "Content Day 5",
              "type": "div",
            },
            "type": "li",
          },
        ],
        "type": "ol",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            currentTemperature={17}
            description="Clear"
            icon="01n"
            location="Newcastle"
/>,
          <ol
            id="tabs"
>
            <li
                        className="tab"
            >
                        <a
                                    href="#day-1"
                        >
                                    Day 1
                        </a>
            </li>
            <li
                        className="tab"
            >
                        <a
                                    href="#day-2"
                        >
                                    Day 2
                        </a>
            </li>
            <li
                        className="tab"
            >
                        <a
                                    href="#day-3"
                        >
                                    Day 3
                        </a>
            </li>
            <li
                        className="tab"
            >
                        <a
                                    href="#day-4"
                        >
                                    Day 4
                        </a>
            </li>
            <li
                        className="tab"
            >
                        <a
                                    href="#day-5"
                        >
                                    Day 5
                        </a>
            </li>
</ol>,
          <ol
            id="tab-panels"
>
            <li
                        className="tab-panel"
                        id="day-1"
            >
                        <div>
                                    Content Day 1
                        </div>
            </li>
            <li
                        className="tab-panel"
                        id="day-2"
            >
                        <div>
                                    Content Day 2
                        </div>
            </li>
            <li
                        className="tab-panel"
                        id="day-3"
            >
                        <div>
                                    Content Day 3
                        </div>
            </li>
            <li
                        className="tab-panel"
                        id="day-4"
            >
                        <div>
                                    Content Day 4
                        </div>
            </li>
            <li
                        className="tab-panel"
                        id="day-5"
            >
                        <div>
                                    Content Day 5
                        </div>
            </li>
</ol>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentTemperature": 17,
            "description": "Clear",
            "icon": "01n",
            "location": "Newcastle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="tab"
>
                <a
                                href="#day-1"
                >
                                Day 1
                </a>
</li>,
              <li
                className="tab"
>
                <a
                                href="#day-2"
                >
                                Day 2
                </a>
</li>,
              <li
                className="tab"
>
                <a
                                href="#day-3"
                >
                                Day 3
                </a>
</li>,
              <li
                className="tab"
>
                <a
                                href="#day-4"
                >
                                Day 4
                </a>
</li>,
              <li
                className="tab"
>
                <a
                                href="#day-5"
                >
                                Day 5
                </a>
</li>,
            ],
            "id": "tabs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#day-1"
>
                  Day 1
</a>,
                "className": "tab",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Day 1",
                  "href": "#day-1",
                },
                "ref": null,
                "rendered": "Day 1",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#day-2"
>
                  Day 2
</a>,
                "className": "tab",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Day 2",
                  "href": "#day-2",
                },
                "ref": null,
                "rendered": "Day 2",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#day-3"
>
                  Day 3
</a>,
                "className": "tab",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Day 3",
                  "href": "#day-3",
                },
                "ref": null,
                "rendered": "Day 3",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#day-4"
>
                  Day 4
</a>,
                "className": "tab",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Day 4",
                  "href": "#day-4",
                },
                "ref": null,
                "rendered": "Day 4",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#day-5"
>
                  Day 5
</a>,
                "className": "tab",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Day 5",
                  "href": "#day-5",
                },
                "ref": null,
                "rendered": "Day 5",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ol",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="tab-panel"
                id="day-1"
>
                <div>
                                Content Day 1
                </div>
</li>,
              <li
                className="tab-panel"
                id="day-2"
>
                <div>
                                Content Day 2
                </div>
</li>,
              <li
                className="tab-panel"
                id="day-3"
>
                <div>
                                Content Day 3
                </div>
</li>,
              <li
                className="tab-panel"
                id="day-4"
>
                <div>
                                Content Day 4
                </div>
</li>,
              <li
                className="tab-panel"
                id="day-5"
>
                <div>
                                Content Day 5
                </div>
</li>,
            ],
            "id": "tab-panels",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Content Day 1
</div>,
                "className": "tab-panel",
                "id": "day-1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Content Day 1",
                },
                "ref": null,
                "rendered": "Content Day 1",
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Content Day 2
</div>,
                "className": "tab-panel",
                "id": "day-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Content Day 2",
                },
                "ref": null,
                "rendered": "Content Day 2",
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Content Day 3
</div>,
                "className": "tab-panel",
                "id": "day-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Content Day 3",
                },
                "ref": null,
                "rendered": "Content Day 3",
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Content Day 4
</div>,
                "className": "tab-panel",
                "id": "day-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Content Day 4",
                },
                "ref": null,
                "rendered": "Content Day 4",
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Content Day 5
</div>,
                "className": "tab-panel",
                "id": "day-5",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Content Day 5",
                },
                "ref": null,
                "rendered": "Content Day 5",
                "type": "div",
              },
              "type": "li",
            },
          ],
          "type": "ol",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
